content/en/blog/_posts/2023-12-07-a-demo-of-webassembly-and-wasi-on-kubernetes/index.md

```diff
@@ -0,0 +1,858 @@
+
+Send a request to the service.
+
+```bash
+curl http://localhost:8080
+```
+
+The output is:
+
+```bash
+Hello, World!
+```
+
+To delete the container, you can execute the following command.
+
+```bash
+ctr task kill wasm-demo-app --signal SIGKILL
+```
+
+### Run Wasm modules on Container Management Platforms
+
+#### Run Wasm modules on Docker Desktop
+
+Docker Desktop also uses runwasi to support the Wasm module. To run Wasm modules within Docker Desktop, ensure the following two options are enabled:
+- Use containerd for pulling and storing images
+- Enable Wasm
+
+{{< figure src="06-enable-wasm-support-on-docker-desktop.png" alt="The diagram shows how to enable Wasm support on Docker Desktop." caption="Figure 6: Enable Wasm support on Docker Desktop" >}}
+
+Click **Apply & restart** to apply the updates. Docker Desktop downloads and installs the following runtimes that can be used to run Wasm modules:
+- `io.containerd.slight.v1`
+- `io.containerd.spin.v1`
+- `io.containerd.wasmedge.v1`
+- `io.containerd.wasmtime.v1`
+
+Use the following `docker run` command to start a Wasm container on your system. `--runtime=io.containerd.wasmedge.v1` informs the Docker engine that you want to use the Wasm containerd shim instead of the standard Linux container runtime.
+
+```bash
+docker run -d -p 8080:8080 \
+--name=wasm-demo-app \
+--runtime=io.containerd.wasmedge.v1 \
+docker.io/cr7258/wasm-demo-app:v1
+```
+
+Send a request to the service.
+
+```bash
+curl http://localhost:8080
+```
+
+The output is:
+
+```bash
+Hello, World!
+```
+
+To delete the container, you can execute the following command.
+
+```bash
+docker rm -f wasm-demo-app
+```
+
+#### Run Wasm modules on Kubernetes
+
+WebAssembly is [driving the third wave of cloud computing](https://nigelpoulton.com/webassembly-the-future-of-cloud-computing/). As the de facto standard in the realm of container orchestration, Kubernetes continuously evolves to leverage the advantages brought about by WebAssembly.
+
+The major way to run Wasm modules on Kubernetes is adding Wasm support to the Kubernetes nodes. To achieve this, we can either manually install a container runtime build with the Wasm runtime library or use the Kwasm Operator to automate the process.
+
+_[Kind](https://kind.sigs.k8s.io/)_ (Kubernetes in Docker) is a tool for running local Kubernetes clusters using Docker containers as "nodes". To facilitate the experiments, use kind to create Kubernetes cluster in the following sections. Run the following command to install Kind.
+
+```bash
+[ $(uname -m) = x86_64 ] && curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.20.0/kind-linux-amd64
+chmod +x ./kind
+sudo mv ./kind /usr/local/bin/kind
+```
+
+##### Kubernetes + Containerd + Crun
+
+In this section, I will demonstrate the manual installation of crun with the WasmEdge runtime library, and adjust containerd config to use crun as the runtime, enabling Wasm support on the Kubernetes node.
+
+Create a single-node Kubernetes cluster using kind.
```

Comment:
I mentioned here there is only 1 node in the cluster. So we don't need to add node selector here?

Reply:
You can instead explain that this advice only makes sense in this specific context and won't be generally applicable. Pick whichever you like of those 2 options (or suggest another). The aim is to make sure that readers don't infer something unhelpful.

Reply:
Ok. I add nodeSelector for the pod: https://github.com/kubernetes/website/pull/43760/commits/4d7edac253c031fbff397678d467cbd64d7e562d